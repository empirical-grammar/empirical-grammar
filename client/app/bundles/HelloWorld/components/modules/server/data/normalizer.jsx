'use strict'

 import React from 'react'
 import _ from 'underscore'

 export default function () {

  var _replace = function (obj, oldField, newField, newValue) {
    var newObject = {}
    newObject[newField] = newValue;
    var newObject = _.chain(newObject).merge(obj).omit(oldField).value();
    return newObject
  }

  var _getValue = function (object, oldField) {
    var oldVal, newVal;
    oldVal = object[oldField]
    if (oldVal instanceof Array) {
      newVal = _.pluck(oldVal, 'id');
    } else {
      newVal = oldVal.id;
    }
    return newVal;
  }

  var _normalizeField = function (object, fieldData) {
    var fieldName = fieldData.name;
    var fieldIdName = fieldData.idName;
    var value = _getValue(object, fieldName)
    var newObject = _replace(object, fieldName, fieldIdName, value)
    return newObject
  }


  // PUBLIC
  this.process = function (object, fieldDatas) {
    return _.reduce(fieldDatas, _normalizeField, object);
  }

}
