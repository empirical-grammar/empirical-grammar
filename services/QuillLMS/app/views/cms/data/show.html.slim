.container
  article.activity-form.activity-data.simple-rounded-box
    form.box-full-form.form-vertical data-redirect=cms_activities_path(@activity.classification.key, flag: @activity.flag) action="/"
      .l-section
        h4 Vendor properties
        iframe#activity-iframe scrolling="no" src=@activity.form_url
      button.btn.btn-primary Save