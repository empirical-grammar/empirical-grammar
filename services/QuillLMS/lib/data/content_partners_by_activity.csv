Tool,UID,Activity ID,Content Partner,Activity Name
Connect,-LvGsoTdi-HFa4r9eNgK,1095,College Board,“The Red Fox Fur Coat”
Connect,-M-H0DCwYDcD3OsfuXWY,1195,College Board,"""The First Day"""
Connect,-M03YCpAMJn8IxSPZKPa,1222,College Board,"""An Occurrence at Owl Creek Bridge"""
Connect,-M03YpZB2gHlSsxS542O,1223,College Board,Lamb to the Slaughter
Connect,-M03WbwMR1ZaXMAFLXro,1231,College Board,What Happened During the Ice Storm
Connect,ffb1d005-ff57-4a02-ba25-c744ef795b4c,1271,College Board,The Night Circus
Connect,39da04f2-eae9-456d-a255-a36bf0f565d0,1272,College Board,"“Drowning in Dishes, but Finding a Home”"
Connect,2c4518d6-2417-479c-8c6a-782e7692ed41,1273,College Board,The Hamilton Mixtape
Connect,84235cb1-bd7a-4404-a2ac-a47dcf7f23cd,1274,College Board,The Girl Who Fell from the Sky
Connect,b03713db-bced-4890-81e8-6414a0dcf46e,1275,College Board,“The Decline of the American Teenager’s Summer Job”
Connect,39239b1f-7caa-4703-b9ca-d90778eb4872,1277,College Board,Lottery
Connect,ab9f51d3-ccf4-49fc-b621-b30721f5583b,1278,College Board,“Teenagers have Stopped Getting Summer Jobs—Why?”
Connect,bc61293f-b376-437c-bc48-e6b98d4b0d81,1279,College Board,All the Light We Cannot See
Connect,7712a90e-8dd7-4297-bf26-b9747b2a2f86,1280,College Board,Romeo and Juliet
Connect,21eac116-a51a-4ea0-8ce6-ff7aa576fe04,1281,College Board,Tamara's Opus
Connect,2e04ef36-ceea-4544-b7c7-751f89fc2b34,1282,College Board,“What to Do with Kids This Summer? Put ‘Em to Work”
Connect,-M1WgGWcOk22p63fohFK,1283,College Board,1984
Connect,-M1WgDhT2QMzj_vnImab,1284,College Board,"""The Fight"""
Connect,-M1WgLi9ZmrvpyXNoshp,1285,College Board,"The Work You Do, the Person You Are"
Connect,6b69f10b-8d7b-45df-8558-dc3da0a5f140,1290,College Board,Out of My Mind
Connect,-Kbk-mjzTNeqWbvVG9zI,476,Core Knowledge,The Boston Massacre
Connect,-KbvuSGyq6ANjB6cSzpQ,477,Core Knowledge,The Stamp Act
Connect,-Kbvv3WS6vGNkeW7d6dt,478,Core Knowledge,The Boston Tea Party
Proofreader,-Kc-iJqt0TsWgfwYkT6m,481,Core Knowledge,The Sons of Liberty
Proofreader,-KoJ1PFy2DNaRhEtlWex,499,Core Knowledge,George Washington
Proofreader,-KoJ9Lrs-pUNFrzjFTz6,503,Core Knowledge,Sailing with the Stars
Proofreader,-KoJB86Hm3C2_OOo9zA6,504,Core Knowledge,The Cape of Good Hope
Proofreader,-KoJCM7vJTVowwUG_NpH,505,Core Knowledge,Totem Poles
Proofreader,-KqUMM-Akp6DDQlVlAbq,516,Core Knowledge,Spices
Proofreader,-KqUQCCoHDkIggc6BMtU,517,Core Knowledge,Roanoke
Proofreader,-KqU.S.2G2QYq3KWYPHazn,518,Core Knowledge,Santa Maria del Fiore
Proofreader,-KqUXZEm8IdwvR2frMfO,520,Core Knowledge,The School of Athens
Proofreader,-KqYDjR0nWR2gNhyOwWY,521,Core Knowledge,Who was Cervantes?
Proofreader,-KqYHrOHtQSIQ4B2T-wa,523,Core Knowledge,The Sapa Inca
Proofreader,-KqYJAxtdM6HVQi7vHpj,524,Core Knowledge,Native Americans & Horses
Proofreader,-KqYKnytOMErPTLZAgUl,525,Core Knowledge,Godasiyo the Woman Chief
Proofreader,-KqYOlcdRBPIkG6Fe-oc,527,Core Knowledge,Navajo Code Talkers
Proofreader,-KqYRDChl1VQ_QB7rEUL,528,Core Knowledge,Vision Quests
Proofreader,-KqYSjSU.S.SUnLNETJEzg,529,Core Knowledge,What's in a Bean?
Proofreader,-KqYXnbNX5-yKYBF0MHF,530,Core Knowledge,Fruit or Vegetable?
Proofreader,-KqZ1GcvRyn-0x5ph2KS,531,Core Knowledge,Tenochtitlán
Proofreader,-KqdKIZmtEyLE5TTfmwp,532,Core Knowledge,Fossils
Proofreader,-KqdMzxI9fqItxVh_Vgy,533,Core Knowledge,Isabella I of Castile
Proofreader,-KqdjKv54k5iifw2FzDk,534,Core Knowledge,Plants: Bee Gardens
Proofreader,-Kqdo2aBDw7bJ4YIXynn,535,Core Knowledge,Maya Calendar
Proofreader,-KqhXp64K1DKX5EtaSYG,536,Core Knowledge,Plate Tectonics
Proofreader,-KqhpFWpVQubJhqWrkFb,537,Core Knowledge,Rocks
Proofreader,-Kqipm5LV2GUkQYGQ8Lh,538,Core Knowledge,Cactus Photosynthesis
Proofreader,-Kqir7E9_3JftVmWhm7z,539,Core Knowledge,Paul Revere's Ride
Proofreader,-KqixRPyWpTzXtBmZtmm,540,Core Knowledge,Colonial Life in the American Revolution
Proofreader,-Kr1Tc2R9sunf8yHM5xh,541,Core Knowledge,Women in the American Revolution
Proofreader,-Kr1VzAFLsATQbuRJlHd,542,Core Knowledge,Man on the Moon
Proofreader,-Kr1cS7S74s-RHF28y9H,544,Core Knowledge,Mars
Proofreader,-Kr20EDdSpxQm93BJrco,545,Core Knowledge,Mae Jemison
Proofreader,-Kr23P_QLluLsDOeT7MB,547,Core Knowledge,The Silk Road
Proofreader,-Kr24ZZ6uu51bYXhADhG,549,Core Knowledge,Shihuangdi's Search for Immortality
Proofreader,-Kr26YEA35YGuXWE0jzb,550,Core Knowledge,Zheng He and the Pirate
Connect,-KqZdYHq5N9O1OYfzZbl,584,Core Knowledge,The Iroquois Confederacy
Connect,-KqZWcwJndWJBU-U3ehj,585,Core Knowledge,The Importance of the Buffalo
Connect,-KqZh2mwDeSPLCxp2UAT,586,Core Knowledge,Ishi
Connect,-KqZaKPdDL_-tnCUDIlX,587,Core Knowledge,The Dawes Act
Connect,-KqZiV1bnPe48h6gyYss,605,Core Knowledge,Magellan
Connect,-KqcJOzHDYUrioJ6fPlR,606,Core Knowledge,Jamestown
Connect,-KqZ_8tESejeyBmj-_pO,608,Core Knowledge,The Travels of Marco Polo
Connect,-KqZUaz7rINZ8fNssJ6r,613,Core Knowledge,Asteroids
Connect,-KqZR__u2hpJ2z8cIch-,614,Core Knowledge,Ham and Alan Shepard
Connect,-KqZFTXwQ5MCZMH9-bUN,615,Core Knowledge,Eclipses
Connect,-KqiiQTci2gs9zhx2tIA,616,Core Knowledge,Tang Dynasty Inventions
Connect,-KqilM2xWovC-Hudmvjn,617,Core Knowledge,Hui Zong's Defeat
Connect,-Kqisu7eUfGfGvjRvT2W,618,Core Knowledge,The Mongol Invasion of China
Connect,-KqigC-G1kUp_pEeQJjl,619,Core Knowledge,Shihuangdi Unifies China
Connect,-KqYkktoWRhHby2187G8,621,Core Knowledge,The Rock Cycle
Connect,-KqYgUp3xAAJYxZtX2yr,622,Core Knowledge,Fold Mountains
Connect,-KqYd2Nk25H2e3fdkwcs,623,Core Knowledge,Hydrothermal Vents
Connect,-KqcjgL00W0ro-Wr4Bts,627,Core Knowledge,Hernan Cortes
Connect,-KqcdV5JA9uU.S.Y09VvOP,628,Core Knowledge,Maya Mystery
Proofreader,-KqZ1GcvRyn-0x5ph2KS,629,Core Knowledge,Tenochtitlán
Connect,-KqcTJxEiuq07PJzbDIc,630,Core Knowledge,The Royal Road
Connect,-KqZQ1fZyO-Lz5UwooN2,632,Core Knowledge,Vascular and Non-vascular Plants
Connect,-KqZLtzic921c7eDVc1D,633,Core Knowledge,Plants in the Food Chain
Connect,-KqZIkV8V3_92A2ajc4u,634,Core Knowledge,Plant Cells and Animal Cells
Connect,-KqZECh5ZxntzLDB-xQK,635,Core Knowledge,Parts of Vascular Plants
Connect,-KqZpAZ8UMhwZp_3ezqw,640,Core Knowledge,Renaissance Beginnings in Italy
Connect,-KqZvArG8422vhwB7rIo,641,Core Knowledge,The Medici Family
Connect,-KqZldiOXNWbJNFvIRY-,642,Core Knowledge,Humanism
Connect,-KqdGWI1rgvtQBg2vIb0,643,Core Knowledge,The Northern Renaissance
Connect,-KqY7BQmhDn71y-E-s3C,728,Core Knowledge,The War Begins at Lexington and Concord
Connect,-KqYNoENpFXSttDu-xPf,729,Core Knowledge,Turning Point: Bunker Hill
Connect,-KqYPpC3o3JeeuhQadBF,730,Core Knowledge,Turning Point: Saratoga
Connect,-KqYK5A0fNos0ZbUjjOu,731,Core Knowledge,The War Ends at Yorktown
Proofreader,-L8ZGcyIM49GOsX_uOPN,747,Core Knowledge,Sputnik 1
Connect,-LAo-erEroO6r6XV1tAV,756,Core Knowledge,International Space Station
Connect,-KhO2hv2Wl8rVCHp1B1G,860,Core Knowledge,Hernando de Soto