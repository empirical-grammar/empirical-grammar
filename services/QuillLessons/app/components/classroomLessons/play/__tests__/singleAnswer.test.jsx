import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import SingleAnswer, { PROJECT } from '../singleAnswer';
import { singleAnswerProps } from './data'

const submissions = {
  student:
    {
      data: "Blah",
      timestamp: "2020-06-15T20:02:41.677Z"
    }
  }

const selectedSubmissionOrder = ['student']

const selectedSubmissions = { student: true }

describe('SingleAnswer component', () => {
  describe('student view', () => {
    Object.defineProperty(window, 'location', {
      value: {
        href: "http://localhost:8090/#/play/class-lessons/-KsJlSHlJ9G0rlVlRIfk?&classroom_unit_id=prvw-dc21ddd3-c766-4a57-8523-3937d5de6c30&student=student"
      }
    });

    describe('inactive', () => {
      it('renders', () => {
        const wrapper = mount(<SingleAnswer {...singleAnswerProps} />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })
    })

    describe('when the student has submitted an answer', () => {
      it('renders', () => {
        const wrapper = mount(<SingleAnswer {...singleAnswerProps} submissions={submissions} />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })
    })

    describe('when an answer is being displayed', () => {
      it('renders', () => {
        const wrapper = mount(<SingleAnswer {...singleAnswerProps} mode={PROJECT} selected_submissions={selectedSubmissions} selected_submission_order={selectedSubmissionOrder} submissions={submissions} />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })
    })
  })

  describe('projector view', () => {
    Object.defineProperty(window, 'location', {
      value: {
        href: "http://localhost:8090/#/play/class-lessons/-KsJlSHlJ9G0rlVlRIfk?&classroom_unit_id=prvw-dc21ddd3-c766-4a57-8523-3937d5de6c30&projector=true"
      }
    });

    describe('inactive', () => {
      it('renders', () => {
        const wrapper = mount(<SingleAnswer {...singleAnswerProps} />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })
    })

    describe('when an answer is being displayed', () => {
      it('renders', () => {
        const wrapper = mount(<SingleAnswer {...singleAnswerProps} mode={PROJECT} selected_submissions={selectedSubmissions} selected_submission_order={selectedSubmissionOrder} submissions={submissions} />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })
    })
  })
})
