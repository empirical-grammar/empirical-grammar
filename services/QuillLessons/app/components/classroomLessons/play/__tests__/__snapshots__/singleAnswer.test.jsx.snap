// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleAnswer component projector view inactive renders 1`] = `
<SingleAnswer
  data={
    Object {
      "play": Object {
        "cues": Array [
          "",
        ],
        "instructions": "Complete the sentence.",
        "prompt": "<p>Yesterday, riding my bike through the town.</p>",
      },
      "teach": Object {
        "script": Array [
          Object {
            "data": Object {
              "body": "<p><strong>Say:</strong> You’ve seen two ways you can correct an incomplete sentence that has an -ing word. Now try correcting one with your partner. You can use either method as long as you write a complete sentence.&nbsp;</p>",
              "heading": "Ask pairs to complete the sentence.",
            },
            "type": "STEP-HTML",
          },
          Object {
            "type": "T-REVIEW",
          },
          Object {
            "data": Object {
              "body": "<p><em>Read the responses out loud.</em></p>
<p><strong>Say: </strong>Are there any responses that are still incomplete sentences?</p>",
              "heading": "Select 2-3 correct responses and 1-2 incorrect responses to display and discuss.",
            },
            "type": "STEP-HTML",
          },
          Object {
            "data": Object {
              "body": "<p><em>Ask the following questions about the incorrect responses:</em></p>
<ul>
  <li>How could we revise this sentence to make it a complete sentence?</li>
  <li>What information is needed to finish the idea?</li>
  <li>Is there anything else we need to change to make the sentence correct?</li>
</ul>",
              "heading": "Discuss responses that are incomplete sentences.",
            },
            "type": "STEP-HTML",
          },
        ],
        "title": "Paired Practice",
      },
    }
  }
  mode={null}
  selected_submission_order={null}
  selected_submissions={null}
  studentCount={1}
  submissions={null}
>
  <div
    className="single-answer "
  >
    <SentenceFragments
      prompt="<p>Yesterday, riding my bike through the town.</p>"
    >
      <div
        className="draft-js sentence-fragments prevent-selection"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Yesterday, riding my bike through the town.</p>",
          }
        }
      />
    </SentenceFragments>
    <_default
      displayArrowAndText={false}
      getQuestion={[Function]}
    >
      <div />
    </_default>
    <Feedback
      feedback={
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Complete the sentence.",
            }
          }
        />
      }
      feedbackType="default"
    >
      <div
        className="student-feedback-container default"
        role="status"
      >
        <div
          className="feedback-row student-feedback-inner-container"
        >
          <img
            alt="Directions Icon"
            className="info"
            src="undefined/images/icons/direction.svg"
          />
          <p
            dangerouslySetInnerHTML={
              Object {
                "__html": "Complete the sentence.",
              }
            }
          />
        </div>
      </div>
    </Feedback>
    <RenderTextEditor
      checkAnswer={[Function]}
      defaultValue=""
      handleChange={[Function]}
      placeholder="Type your response here"
    >
      <div
        className="student text-editor card is-fullwidth  "
      >
        <div
          className="card-content"
        >
          <div
            className="content"
          >
            <TextareaAutosize
              autoCapitalize="off"
              autoCorrect="off"
              autoFocus={true}
              className="connect-text-area"
              onChange={[Function]}
              onHeightChange={[Function]}
              onInput={[Function]}
              onKeyDown={[Function]}
              placeholder="Type your response here"
              spellCheck={false}
              useCacheForDOMMeasurements={false}
            >
              <textarea
                autoCapitalize="off"
                autoCorrect="off"
                autoFocus={true}
                className="connect-text-area"
                onChange={[Function]}
                onInput={[Function]}
                onKeyDown={[Function]}
                placeholder="Type your response here"
                spellCheck={false}
                style={
                  Object {
                    "height": 0,
                  }
                }
              />
            </TextareaAutosize>
          </div>
        </div>
      </div>
    </RenderTextEditor>
    <div
      className="question-button-group"
    >
      <button
        className="quill-button primary contained large focus-on-light disabled"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
</SingleAnswer>
`;

exports[`SingleAnswer component projector view when an answer is being displayed renders 1`] = `
<SingleAnswer
  data={
    Object {
      "play": Object {
        "cues": Array [
          "",
        ],
        "instructions": "Complete the sentence.",
        "prompt": "<p>Yesterday, riding my bike through the town.</p>",
      },
      "teach": Object {
        "script": Array [
          Object {
            "data": Object {
              "body": "<p><strong>Say:</strong> You’ve seen two ways you can correct an incomplete sentence that has an -ing word. Now try correcting one with your partner. You can use either method as long as you write a complete sentence.&nbsp;</p>",
              "heading": "Ask pairs to complete the sentence.",
            },
            "type": "STEP-HTML",
          },
          Object {
            "type": "T-REVIEW",
          },
          Object {
            "data": Object {
              "body": "<p><em>Read the responses out loud.</em></p>
<p><strong>Say: </strong>Are there any responses that are still incomplete sentences?</p>",
              "heading": "Select 2-3 correct responses and 1-2 incorrect responses to display and discuss.",
            },
            "type": "STEP-HTML",
          },
          Object {
            "data": Object {
              "body": "<p><em>Ask the following questions about the incorrect responses:</em></p>
<ul>
  <li>How could we revise this sentence to make it a complete sentence?</li>
  <li>What information is needed to finish the idea?</li>
  <li>Is there anything else we need to change to make the sentence correct?</li>
</ul>",
              "heading": "Discuss responses that are incomplete sentences.",
            },
            "type": "STEP-HTML",
          },
        ],
        "title": "Paired Practice",
      },
    }
  }
  mode="PROJECT"
  selected_submission_order={
    Array [
      "student",
    ]
  }
  selected_submissions={
    Object {
      "student": true,
    }
  }
  studentCount={1}
  submissions={
    Object {
      "student": Object {
        "data": "Blah",
        "timestamp": "2020-06-15T20:02:41.677Z",
      },
    }
  }
>
  <div
    className="single-answer "
  >
    <SentenceFragments
      prompt="<p>Yesterday, riding my bike through the town.</p>"
    >
      <div
        className="draft-js sentence-fragments prevent-selection"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Yesterday, riding my bike through the town.</p>",
          }
        }
      />
    </SentenceFragments>
    <div
      className="display-answers-divider prompt-showing"
    >
      <div
        className="display-answers-divider-content"
      >
        <button
          className="focus-on-light"
          onClick={[Function]}
          type="button"
        >
          Hide
        </button>
      </div>
    </div>
    <div
      className="display-mode"
    >
      <div>
        <p
          className="answer-header"
        >
          Your response
        </p>
        <p
          className="your-answer"
        >
          Blah
        </p>
      </div>
      <div>
        <p
          className="answer-header"
        >
          Class responses
        </p>
        <ol
          className="class-answer-list"
        >
          <li
            key="0"
          >
            Blah
          </li>
        </ol>
      </div>
    </div>
  </div>
</SingleAnswer>
`;

exports[`SingleAnswer component student view inactive renders 1`] = `
<SingleAnswer
  data={
    Object {
      "play": Object {
        "cues": Array [
          "",
        ],
        "instructions": "Complete the sentence.",
        "prompt": "<p>Yesterday, riding my bike through the town.</p>",
      },
      "teach": Object {
        "script": Array [
          Object {
            "data": Object {
              "body": "<p><strong>Say:</strong> You’ve seen two ways you can correct an incomplete sentence that has an -ing word. Now try correcting one with your partner. You can use either method as long as you write a complete sentence.&nbsp;</p>",
              "heading": "Ask pairs to complete the sentence.",
            },
            "type": "STEP-HTML",
          },
          Object {
            "type": "T-REVIEW",
          },
          Object {
            "data": Object {
              "body": "<p><em>Read the responses out loud.</em></p>
<p><strong>Say: </strong>Are there any responses that are still incomplete sentences?</p>",
              "heading": "Select 2-3 correct responses and 1-2 incorrect responses to display and discuss.",
            },
            "type": "STEP-HTML",
          },
          Object {
            "data": Object {
              "body": "<p><em>Ask the following questions about the incorrect responses:</em></p>
<ul>
  <li>How could we revise this sentence to make it a complete sentence?</li>
  <li>What information is needed to finish the idea?</li>
  <li>Is there anything else we need to change to make the sentence correct?</li>
</ul>",
              "heading": "Discuss responses that are incomplete sentences.",
            },
            "type": "STEP-HTML",
          },
        ],
        "title": "Paired Practice",
      },
    }
  }
  mode={null}
  selected_submission_order={null}
  selected_submissions={null}
  studentCount={1}
  submissions={null}
>
  <div
    className="single-answer "
  >
    <SentenceFragments
      prompt="<p>Yesterday, riding my bike through the town.</p>"
    >
      <div
        className="draft-js sentence-fragments prevent-selection"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Yesterday, riding my bike through the town.</p>",
          }
        }
      />
    </SentenceFragments>
    <_default
      displayArrowAndText={false}
      getQuestion={[Function]}
    >
      <div />
    </_default>
    <Feedback
      feedback={
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Complete the sentence.",
            }
          }
        />
      }
      feedbackType="default"
    >
      <div
        className="student-feedback-container default"
        role="status"
      >
        <div
          className="feedback-row student-feedback-inner-container"
        >
          <img
            alt="Directions Icon"
            className="info"
            src="undefined/images/icons/direction.svg"
          />
          <p
            dangerouslySetInnerHTML={
              Object {
                "__html": "Complete the sentence.",
              }
            }
          />
        </div>
      </div>
    </Feedback>
    <RenderTextEditor
      checkAnswer={[Function]}
      defaultValue=""
      handleChange={[Function]}
      placeholder="Type your response here"
    >
      <div
        className="student text-editor card is-fullwidth  "
      >
        <div
          className="card-content"
        >
          <div
            className="content"
          >
            <TextareaAutosize
              autoCapitalize="off"
              autoCorrect="off"
              autoFocus={true}
              className="connect-text-area"
              onChange={[Function]}
              onHeightChange={[Function]}
              onInput={[Function]}
              onKeyDown={[Function]}
              placeholder="Type your response here"
              spellCheck={false}
              useCacheForDOMMeasurements={false}
            >
              <textarea
                autoCapitalize="off"
                autoCorrect="off"
                autoFocus={true}
                className="connect-text-area"
                onChange={[Function]}
                onInput={[Function]}
                onKeyDown={[Function]}
                placeholder="Type your response here"
                spellCheck={false}
                style={
                  Object {
                    "height": 0,
                  }
                }
              />
            </TextareaAutosize>
          </div>
        </div>
      </div>
    </RenderTextEditor>
    <div
      className="question-button-group"
    >
      <button
        className="quill-button primary contained large focus-on-light disabled"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
</SingleAnswer>
`;

exports[`SingleAnswer component student view when an answer is being displayed renders 1`] = `
<SingleAnswer
  data={
    Object {
      "play": Object {
        "cues": Array [
          "",
        ],
        "instructions": "Complete the sentence.",
        "prompt": "<p>Yesterday, riding my bike through the town.</p>",
      },
      "teach": Object {
        "script": Array [
          Object {
            "data": Object {
              "body": "<p><strong>Say:</strong> You’ve seen two ways you can correct an incomplete sentence that has an -ing word. Now try correcting one with your partner. You can use either method as long as you write a complete sentence.&nbsp;</p>",
              "heading": "Ask pairs to complete the sentence.",
            },
            "type": "STEP-HTML",
          },
          Object {
            "type": "T-REVIEW",
          },
          Object {
            "data": Object {
              "body": "<p><em>Read the responses out loud.</em></p>
<p><strong>Say: </strong>Are there any responses that are still incomplete sentences?</p>",
              "heading": "Select 2-3 correct responses and 1-2 incorrect responses to display and discuss.",
            },
            "type": "STEP-HTML",
          },
          Object {
            "data": Object {
              "body": "<p><em>Ask the following questions about the incorrect responses:</em></p>
<ul>
  <li>How could we revise this sentence to make it a complete sentence?</li>
  <li>What information is needed to finish the idea?</li>
  <li>Is there anything else we need to change to make the sentence correct?</li>
</ul>",
              "heading": "Discuss responses that are incomplete sentences.",
            },
            "type": "STEP-HTML",
          },
        ],
        "title": "Paired Practice",
      },
    }
  }
  mode="PROJECT"
  selected_submission_order={
    Array [
      "student",
    ]
  }
  selected_submissions={
    Object {
      "student": true,
    }
  }
  studentCount={1}
  submissions={
    Object {
      "student": Object {
        "data": "Blah",
        "timestamp": "2020-06-15T20:02:41.677Z",
      },
    }
  }
>
  <div
    className="single-answer "
  >
    <SentenceFragments
      prompt="<p>Yesterday, riding my bike through the town.</p>"
    >
      <div
        className="draft-js sentence-fragments prevent-selection"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Yesterday, riding my bike through the town.</p>",
          }
        }
      />
    </SentenceFragments>
    <div
      className="display-answers-divider prompt-showing"
    >
      <div
        className="display-answers-divider-content"
      >
        <button
          className="focus-on-light"
          onClick={[Function]}
          type="button"
        >
          Hide
        </button>
      </div>
    </div>
    <div
      className="display-mode"
    >
      <div>
        <p
          className="answer-header"
        >
          Your response
        </p>
        <p
          className="your-answer"
        >
          Blah
        </p>
      </div>
      <div>
        <p
          className="answer-header"
        >
          Class responses
        </p>
        <ol
          className="class-answer-list"
        >
          <li
            key="0"
          >
            Blah
          </li>
        </ol>
      </div>
    </div>
  </div>
</SingleAnswer>
`;

exports[`SingleAnswer component student view when the student has submitted an answer renders 1`] = `
<SingleAnswer
  data={
    Object {
      "play": Object {
        "cues": Array [
          "",
        ],
        "instructions": "Complete the sentence.",
        "prompt": "<p>Yesterday, riding my bike through the town.</p>",
      },
      "teach": Object {
        "script": Array [
          Object {
            "data": Object {
              "body": "<p><strong>Say:</strong> You’ve seen two ways you can correct an incomplete sentence that has an -ing word. Now try correcting one with your partner. You can use either method as long as you write a complete sentence.&nbsp;</p>",
              "heading": "Ask pairs to complete the sentence.",
            },
            "type": "STEP-HTML",
          },
          Object {
            "type": "T-REVIEW",
          },
          Object {
            "data": Object {
              "body": "<p><em>Read the responses out loud.</em></p>
<p><strong>Say: </strong>Are there any responses that are still incomplete sentences?</p>",
              "heading": "Select 2-3 correct responses and 1-2 incorrect responses to display and discuss.",
            },
            "type": "STEP-HTML",
          },
          Object {
            "data": Object {
              "body": "<p><em>Ask the following questions about the incorrect responses:</em></p>
<ul>
  <li>How could we revise this sentence to make it a complete sentence?</li>
  <li>What information is needed to finish the idea?</li>
  <li>Is there anything else we need to change to make the sentence correct?</li>
</ul>",
              "heading": "Discuss responses that are incomplete sentences.",
            },
            "type": "STEP-HTML",
          },
        ],
        "title": "Paired Practice",
      },
    }
  }
  mode={null}
  selected_submission_order={null}
  selected_submissions={null}
  studentCount={1}
  submissions={
    Object {
      "student": Object {
        "data": "Blah",
        "timestamp": "2020-06-15T20:02:41.677Z",
      },
    }
  }
>
  <div
    className="single-answer "
  >
    <SentenceFragments
      prompt="<p>Yesterday, riding my bike through the town.</p>"
    >
      <div
        className="draft-js sentence-fragments prevent-selection"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Yesterday, riding my bike through the town.</p>",
          }
        }
      />
    </SentenceFragments>
    <_default
      displayArrowAndText={false}
      getQuestion={[Function]}
    >
      <div />
    </_default>
    <Feedback
      feedback={
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "Complete the sentence.",
            }
          }
        />
      }
      feedbackType="default"
    >
      <div
        className="student-feedback-container default"
        role="status"
      >
        <div
          className="feedback-row student-feedback-inner-container"
        >
          <img
            alt="Directions Icon"
            className="info"
            src="undefined/images/icons/direction.svg"
          />
          <p
            dangerouslySetInnerHTML={
              Object {
                "__html": "Complete the sentence.",
              }
            }
          />
        </div>
      </div>
    </Feedback>
    <RenderTextEditor
      checkAnswer={[Function]}
      defaultValue=""
      disabled={true}
      handleChange={[Function]}
      placeholder="Type your response here"
      value="Blah"
    >
      <div
        className="student text-editor card is-fullwidth  disabled-editor"
      >
        <div
          className="card-content"
        >
          <div
            className="content"
          >
            <TextareaAutosize
              autoCapitalize="off"
              autoCorrect="off"
              autoFocus={true}
              className="connect-text-area"
              disabled={true}
              onChange={[Function]}
              onHeightChange={[Function]}
              onInput={[Function]}
              onKeyDown={[Function]}
              placeholder="Type your response here"
              spellCheck={false}
              useCacheForDOMMeasurements={false}
              value="Blah"
            >
              <textarea
                autoCapitalize="off"
                autoCorrect="off"
                autoFocus={true}
                className="connect-text-area"
                disabled={true}
                onChange={[Function]}
                onInput={[Function]}
                onKeyDown={[Function]}
                placeholder="Type your response here"
                spellCheck={false}
                style={
                  Object {
                    "height": 0,
                  }
                }
                value="Blah"
              />
            </TextareaAutosize>
          </div>
        </div>
      </div>
    </RenderTextEditor>
    <div
      className="submitted-bar"
    >
      Please wait as your teacher reviews your response.
    </div>
  </div>
</SingleAnswer>
`;
